app/build.gradle

```diff
@@ -77,18 +77,33 @@ def getLastFMApiKey() {
 
 dependencies {
 // Support libraries
+    implementation 'androidx.annotation:annotation:1.0.2'
     implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.arch.core:core-common:2.1.0'
+    implementation 'androidx.cardview:cardview:1.0.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
-    implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
+    implementation 'androidx.coordinatorlayout:coordinatorlayout:1.0.0'
+    implementation 'androidx.core:core:1.0.2'
+    implementation 'androidx.drawerlayout:drawerlayout:1.0.0'
+    implementation 'androidx.fragment:fragment:1.0.0'
     annotationProcessor 'androidx.lifecycle:lifecycle-compiler:2.0.0'
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
+    implementation 'androidx.lifecycle:lifecycle-livedata-core:2.0.0'
+    implementation 'androidx.lifecycle:lifecycle-viewmodel:2.0.0'
+    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0' //DEPRECATED
+    implementation 'androidx.media:media:1.0.1' //Replace with androidx.media2
     implementation 'androidx.mediarouter:mediarouter:1.0.0'
     implementation 'androidx.multidex:multidex:2.0.1'
+    implementation 'androidx.paging:paging-common:2.1.0'
     implementation 'androidx.paging:paging-runtime:2.1.0'
     implementation 'androidx.preference:preference:1.1.0-rc01'
     implementation 'androidx.recyclerview:recyclerview:1.0.0'
-    implementation 'androidx.room:room-runtime:2.1.0'
+    implementation 'androidx.room:room-common:2.1.0'
     annotationProcessor 'androidx.room:room-compiler:2.1.0'
+    implementation 'androidx.room:room-runtime:2.1.0'
+    implementation 'androidx.sqlite:sqlite:2.0.1'
+    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.0.0'
```

Comment:
I'm not entirely sure where new dependencies appeared from. Could you elaborate some more?

Reply:
I just added all dependencies that had at least one usage. For example, androidx.sqlite is used in RadioDroidDatabase.java, but wasn't in dependencies (as is the case with the other new dependencies). I think it's best to keep track of all AndroidX usages, because sometimes Studio likes to automatically upgrade AndroidX versions that weren't explicitly written (with a certain version) in dependencies, and that could have the side effect of breaking stuff.

Reply:
Let's take `room-runtime` for example, https://mvnrepository.com/artifact/androidx.room/room-runtime/2.1.0  it depends on `androidx.sqlite` the version of which is set in stone and would not undergo random update, so there is no point in explicitly listing it.

Reply:
Correct me if I'm wrong, but isn't sqlite in this specific file being used separately (or a separate implementation) from room-runtime?

Reply:
Sorry, I don't understand your question.

`room-runtime` depends on `androidx.sqlite` so `androidx.sqlite` is being pulled as dependency the same way `room-runtime` is pulled but just implicitly. So `RadioDroidDatabase.java` is using `androidx.sqlite` which is pulled by `room-runtime`. 

Android Studio cannot get package without it being either explicitly listed in gradle or implicitly as a dependency of other package.